packages/coding-standards/src/Phpstan/OrmPropertyGetterAndSetterHasNoTypehintRule.php

```diff
@@ -110,4 +172,20 @@ private function isMethodInAncestor(string $methodName, Scope $scope): bool
 
         return false;
     }
+
+    /**
+     * @param string $parameterType
+     * @param \PHPStan\Analyser\Scope $scope
+     * @return bool
+     */
+    private function isParameterDataObjectOfCurrentEntity(string $parameterType, Scope $scope): bool
+    {
+        $currentClassName = $scope->getClassReflection()?->getName();
+
+        if ($currentClassName === null) {
+            return false;
+        }
+
+        return str_ends_with($parameterType, 'Data') && str_starts_with($parameterType, $currentClassName);
```

Comment:
```suggestion
        return str_starts_with($parameterType, $currentClassName . 'Data');
```

Your way it would pass for Product entity when ProductVideoData would be provided.

Reply:
good catch, thanks! fixed in 5b97590957